Composer/packages/client/src/components/Auth/AuthDialog.tsx

```diff
@@ -9,17 +9,21 @@ import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
 import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button';
 import { TextField } from 'office-ui-fabric-react/lib/TextField';
 import { useCallback, useState } from 'react';
+import { useRecoilValue } from 'recoil';
 
+import { dispatcherState } from '../../recoilModel/atoms';
 import storage from '../../utils/storage';
 import { isTokenExpired } from '../../utils/auth';
 
 export interface AuthDialogProps {
   needGraph: boolean;
   onDismiss: () => void;
-  next: () => void;
+  next?: () => void;
 }
 export const AuthDialog: React.FC<AuthDialogProps> = (props) => {
-  const [graphToken, setGraphToken] = useState('');
+  const { setPrimaryToken, setGraphToken } = useRecoilValue(dispatcherState);
+
```

Comment:
Why is there a local state and recoil state graph token. Cant we just use the recoils state graph token and remove the local one that responds to it?

Reply:
Oh nvm I see, Its beause you do not set the recoil state one until the continue button is clicked. disregard this comment.